### ISSUE TITLE
feature request: automatically PR meeting minutes once meeting has concluded

### ISSUE THREAD

**Issue by bnb** [pkgjs/meet#10]
At present there's several manual steps of automatically PR'ing in the meeting minutes. Need to copy from Google Docs, create a new file in the repo with the correct title structure (currently, most repos use the date of the meeting as the filename) in the correct directory, and submit a PR. This is a process that unnecessarily uses human time that could be used elsewhere for more productive things.

It would be excellent if meet could automatically PR any meeting minutes for meetings that have already completed but have not yet been PR'ed.

**Comment by mmarchini** [comment]
Maybe this could be a separate Action (separation of concerns)?

**Comment by bnb** [comment]
It definitely could be! It would potentially be nice to have an all-in-one that reduces the overhead of configuration and setup for folks trying to ramp up to the tooling and just get going, but I could conceive of reasons to have it be separate.

**Comment by mmarchini** [comment]
I guess I'd be fine with it being implemented either way :)

**Comment by bnb** [comment]
I would also be fine with it being implemented either way - if someone's willing to do the work, whatever they're willing/able to do is a +1 from me :)

**Comment by mmarchini** [comment]
I'm sorta looking into it, although I'm writing an example using HackMD instead of Google Drive (I suggested we try out HackMD on nodejs/diagnostics). One thing we would need to decide is how to fetch the meeting issue and how to find the meeting link in the issue. We could either use a heuristic approach (search issues by title name and chose the most recently opened meeting + search the issue body for "Minutes: .*"), or we could have explicit identifiers to make things easier (have a label `current-meeting` or `next-meeting`, and in the issue we could add the link to the minutes as metadata as a comment `<!-- minutes: .* -->`). The upside of the latter approach is that if title or the issue text is manually changed, automation will still works).

Another thing we can do as a part of this feature is to add the recording link to the minutes (which has similar challenges as suggested above, plus finding the recording which I think is harder than finding an issue).

**Comment by bnb** [comment]
I'm in favor of the label approach since it also provides a good way for humans to identify which meeting issue is the one they should be looking at (some repos are slow to close old meeting issues and the title system is not currently very helpful).

@mmarchini could you explain the metadata comment a bit more to me? I don't totally understand.

**Comment by mmarchini** [comment]
I think placeholder is a better word than metadata for what I meant. Basically, instead of having:

```markdown
## Links

* **Recording**: 
* **GitHub Issue**: https://github.com/nodejs/diagnostics/issues/397
* **Minutes Google Doc**: https://docs.google.com/document/d/14W5g4rHw38XzX1wGYgCOSXY-v7AW0_jw_b2IcctnGKM/edit
```

We would have

```markdown
## Links

* **Recording**: <!-- recording-placeholder -->
* **GitHub Issue**: https://github.com/nodejs/diagnostics/issues/397
* **Minutes Google Doc**: https://docs.google.com/document/d/14W5g4rHw38XzX1wGYgCOSXY-v7AW0_jw_b2IcctnGKM/edit
<!-- minutes:https://docs.google.com/document/d/14W5g4rHw38XzX1wGYgCOSXY-v7AW0_jw_b2IcctnGKM/edit -->
```

Since `<!-- -->` are comments, they are not rendered, but are easier to be found by scripts.

**Comment by bnb** [comment]
Another feature that may be potentially useful and completely solve this: https://hackmd.io/c/tutorials/%2Fs%2Flink-with-github

**Comment by mmarchini** [comment]
Right, as long as we don't require PRs for minutes it should work. Reading the comments it I wonder how well it would work with several folks making changes to the minutes and trying to push (because users need to push and pull manually). Also, seems like this feature requires users to manually link the minute every time (unless there's an API for that, which I don't think there is yet).

Worth keeping an eye on this feature, they might add improvements to it in the future :)

**Comment by bnb** [comment]
cc @joesepi this is the issue for the feature you want, by the way. Would genuinely love to see a PR around this - shouldn't be massively complex ❤️
